core/com.b2international.snowowl.datastore/src/com/b2international/snowowl/datastore/CodeSystemVersionEntry.java

```diff
@@ -27,17 +32,63 @@
 import com.b2international.snowowl.core.date.EffectiveTimes;
 import com.b2international.snowowl.datastore.cdo.CDOIDUtils;
 import com.b2international.snowowl.terminologymetadata.CodeSystemVersion;
+import com.b2international.snowowl.terminologymetadata.TerminologymetadataPackage;
 import com.fasterxml.jackson.annotation.JsonIgnore;
 import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
 import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
+import com.google.common.primitives.Longs;
 
 /**
  * CDO independent representation of a {@link CodeSystemVersion}.
  */
 @Doc
 @JsonDeserialize(builder = CodeSystemVersionEntry.Builder.class)
-public class CodeSystemVersionEntry implements ICodeSystemVersion {
+public final class CodeSystemVersionEntry implements Serializable {
 
+	/**
+	 * Unique terminology component identifier for versions.
+	 */
+	public static final short TERMINOLOGY_COMPONENT_ID = 2;
+	
+	/**Constant for {@value} artefacts.*/
+	public static final  String UNVERSIONED = "Unversioned";
+	
+	/**Timestamp indicating that a version branch has not been modified since the corresponding {@link ICodeSystemVersion} has been created.*/
+	public static final long NOT_MODIFIED_YET_LAST_UPDATE_TIME = EffectiveTimes.UNSET_EFFECTIVE_TIME;
+	
+	/**
+	 * This effective time has to be manually set on the {@link TerminologymetadataPackage#getCodeSystemVersion_LastUpdateDate()}
+	 * feature before re-versioning an already existing version.
+	 * <p>This is the workaround to lie to CDO about a change, since the commit timestamp not known at commit time but
+	 * on change processing time.
+	 */
+	public static final Date FAKE_LAST_UPDATE_TIME_DATE = new Date(Dates.MIN_DATE_LONG);
```

Comment:
It might be worth documenting it around this constant that the change processors will always modify the date to the commit timestamp if they find a `CDOSetFeatureDelta` for this property, regardless of whether this or another value is set on the object.